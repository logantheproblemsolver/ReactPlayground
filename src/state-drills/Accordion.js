import React, {Component} from 'react';


class Accordion extends Component {
    static defaultProps = {
        section: []
    };

    state = {
        activeSectionIndex: null,
    }


    handleSetActiveSection= (sectionIndex) => {
        this.setState({
            activeSectionIndex: sectionIndex,
        })
    }


    renderItem(section, idx, activeSectionIndex) {
        return (
            <li key={idx}>
                <button 
                    type='button'
                    onClick={() => this.handleSetActiveSection(idx)}>
                        {section.title}
                </button>
                {(activeSectionIndex === idx) && <p>{section.content}</p>}    
            </li>
        )
    }

    render() {
        const {activeSectionIndex} = this.state
        const {section} = this.props
        return (
            <ul>
                {section.map((section,  idx) => 
                    this.renderItem(section, idx, activeSectionIndex)
                )}
            </ul>
        )
    }
}

export default Accordion