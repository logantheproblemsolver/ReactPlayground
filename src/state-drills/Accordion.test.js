import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';


const section = [
    {
      title: 'Title1',
      content: 'Content1',
    },
    {
      title: 'Title2',
      content: 'Content2',
    },
    {
      title: 'Title3',
      content: 'Content3',
    },
    {
      title: 'Title4',
      content: 'Content4'
    }
  ]


describe('Accordion Component', () => {
    it('renders empty given no Accordion without errors', () => {
        const wrappers = shallow(<Accordion />)
        expect(toJson(wrappers)).toMatchSnapshot()
    })
    it('renders no sections by default', () => {
        const wrappers = shallow(<Accordion section={section} />)
    })
    it('opens any clicked section', () => {
        const wrappers = shallow(<Accordion section={section} />)
        wrappers.find('button').at(1).simulate('click')
        expect(toJson(wrappers)).toMatchSnapshot()
    })
    it('only opens one section at a time', () => {
        const wrapper = shallow(<Accordion section={section} />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})