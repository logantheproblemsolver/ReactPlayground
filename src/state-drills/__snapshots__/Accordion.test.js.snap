// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens one section at a time 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Title1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Title2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Title3
    </button>
    <p>
      Content3
    </p>
  </li>
  <li
    key="3"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Title4
    </button>
  </li>
</ul>
`;

exports[`Accordion Component opens any clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Title1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Title2
    </button>
    <p>
      Content2
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Title3
    </button>
  </li>
  <li
    key="3"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Title4
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no Accordion without errors 1`] = `<ul />`;
